lib/codegen/src/simple_gvn.rs

```diff
@@ -20,64 +22,88 @@ fn trivially_unsafe_for_gvn(opcode: Opcode) -> bool {
         || opcode.writes_cpu_flags()
 }
 
+/// Wrapper around `InstructionData` which implements `Eq` and `Hash`
+#[derive(Clone)]
+struct HashEntry<'a, 'f: 'a>(InstructionData, Type, &'a RefCell<FuncCursor<'f>>);
```

Comment:
Should we call this `HashKey` rather than `HashEntry`, to avoid confusion with `HashMap`'s `entry` concept?